code/modules/library/admin_only.dm

```diff
@@ -0,0 +1,354 @@
+#define BOOK_ADMIN_DELETE "deleted"
+#define BOOK_ADMIN_RESTORE "undeleted"
+#define BOOK_ADMIN_REPORT "reported"
+
+/obj/machinery/computer/libraryconsole/admin_only_do_not_map_in_you_fucker
+	/// When a user clicks view, do we display the raw text, or process it with markdown
+	var/view_raw = FALSE
+	/// If we should show deleted entries or not
+	var/show_deleted = TRUE
+	/// The current ckey we're looking for
+	var/ckey = ""
+	/// List mapping requested book ids to a list of their edit logs
+	var/list/book_history = list()
+	interface_type = "LibraryAdmin"
```

Comment:
old vars before new vars, add this above all the new stuff you implement on this type